opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/config/AuthorizationSessionKeyConfig.java

```diff
@@ -0,0 +1,59 @@
+package de.adorsys.opba.tppauthapi.config;
+
+import de.adorsys.opba.api.security.internal.EnableTokenBasedApiSecurity;
+import de.adorsys.opba.api.security.internal.service.TokenBasedAuthService;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.web.context.WebApplicationContext;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static de.adorsys.opba.restapi.shared.HttpHeaders.AUTHORIZATION_SESSION_KEY;
+
+@Configuration
+@EnableTokenBasedApiSecurity
+@RequiredArgsConstructor
+@Getter
+@Slf4j
+public class AuthorizationSessionKeyConfig {
+
+    private final TokenBasedAuthService authService;
+
+    @Bean
+    @Scope(scopeName = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
+    public AuthorizationSessionKeyFromHttpRequest getAuthorizationSessionKeyFromHttpRequest(HttpServletRequest httpServletRequest) {
+        log.info("REQUEST COMMING IN {}",  httpServletRequest.getRequestURI());
+        if (null == httpServletRequest.getCookies()) {
+            log.info("rest request without authorization {}", httpServletRequest.getRequestURI());
+            return new AuthorizationSessionKeyFromHttpRequest(Optional.empty());
```

Comment:
Instead of using wrapping above optional - explicitly return null

Reply:
done

Reply:
Hm... ok, but maybe we can throw an exception here

Reply:
No, its ok to have no cookie. As explained, as after login page is completly new loaded I dont have any handle to start with. So when page is loaded very first time, I start timer. Timer rings. But at that time user propably has not logged in. so timer call fails and is never renewed.
When user logs in, page is completly new loaded. So timer rings again and than cookie is sent with call and valid. So in server throwing an exception would lead to fail of initialization of PsuAuthController. I just checked in all changes and can demonstrate, if wanted.

Reply:
You can catch 400 sent by server with JS if cookie is absent. Havent found any sever problems with this patch
```
Index: opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/controller/PsuAuthController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/controller/PsuAuthController.java	(revision a5d6287faf9166a2c1f2a2aa588fae759d6c9f3b)
+++ opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/controller/PsuAuthController.java	(date 1588942295000)
@@ -110,11 +110,11 @@
 
     @Override
     public ResponseEntity<Void> renewalAuthorizationSessionKey(UUID xRequestId, UUID authorizationId) {
-        if (!authorizationKeyFromHttpRequest.isPresent()) {
+        /*if (!authorizationKeyFromHttpRequest.isPresent()) {
             log.warn("Authorization session key renewal call failed for {} authorization - missing cookie", authorizationId);
             return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
-        }
-        String[] cookies = buildAuthorizationCookiesOnAllPaths(authorizationId, authorizationKeyFromHttpRequest.get());
+        }*/
+        String[] cookies = buildAuthorizationCookiesOnAllPaths(authorizationId, authorizationKeyFromHttpRequest.getKey());
         String ttl = Long.toString(cookieProperties.getMaxAge().getSeconds());
         log.debug("cookie is renewed for authid {} for time {}", authorizationId, ttl);
         return ResponseEntity
Index: opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/config/AuthorizationSessionKeyConfig.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/config/AuthorizationSessionKeyConfig.java	(revision a5d6287faf9166a2c1f2a2aa588fae759d6c9f3b)
+++ opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/config/AuthorizationSessionKeyConfig.java	(date 1588942240000)
@@ -15,7 +15,6 @@
 import javax.servlet.http.Cookie;
 import javax.servlet.http.HttpServletRequest;
 import java.util.Arrays;
-import java.util.Optional;
 
 import static de.adorsys.opba.restapi.shared.HttpHeaders.AUTHORIZATION_SESSION_KEY;
 
@@ -33,7 +32,7 @@
     public AuthorizationSessionKeyFromHttpRequest getAuthorizationSessionKeyFromHttpRequest(HttpServletRequest httpServletRequest) {
         log.debug("REQUEST COMMING IN {}",  httpServletRequest.getRequestURI());
         if (null == httpServletRequest.getCookies()) {
-            return new AuthorizationSessionKeyFromHttpRequest(Optional.empty());
+            throw new IllegalArgumentException("No cookie");
         }
 
         String authCookieValue = Arrays.stream(httpServletRequest.getCookies())
@@ -42,17 +41,12 @@
                 .map(Cookie::getValue)
                 .orElseThrow(() -> new IllegalStateException("Authorization-Session-Key cookie is missing"));
 
-        return new AuthorizationSessionKeyFromHttpRequest(Optional.of(authService.validateTokenAndGetSubject(authCookieValue)));
+        return new AuthorizationSessionKeyFromHttpRequest(authService.validateTokenAndGetSubject(authCookieValue));
     }
 
+    @Getter
     @AllArgsConstructor
     public static class AuthorizationSessionKeyFromHttpRequest {
-        private Optional<String> key;
-        public String get() {
-            return key.get();
-        }
-        public boolean isPresent() {
-            return key.isPresent();
-        }
+        private final String key;
     }
 }
```
So, probably we need to look at your issue

Reply:
With this patch I get exception, as song, as user is not logged in and timeout is reached:

17:32:55.497 [http-nio-8085-exec-6] DEBUG d.a.o.t.config.AuthorizationSessionKeyConfig - REQUEST COMMING IN /v1/psu/ais/52f8e2bc-53f3-48c1-b399-033b9c7461d2/renewal-authorization-session-key
17:32:55.504 [http-nio-8085-exec-6] ERROR d.a.o.t.controller.GenericControllerAdvice - Unhandled exception: Error creating bean with name 'scopedTarget.getAuthorizationSessionKeyFromHttpRequest' defined in class path resource [de/adorsys/opba/tppauthapi/config/AuthorizationSessionKeyConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [de.adorsys.opba.tppauthapi.config.AuthorizationSessionKeyConfig$AuthorizationSessionKeyFromHttpRequest]: Factory method 'getAuthorizationSessionKeyFromHttpRequest' threw exception; nested exception is java.lang.IllegalStateException: Authorization-Session-Key cookie is missing
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'scopedTarget.getAuthorizationSessionKeyFromHttpRequest' defined in class path resource [de/adorsys/opba/tppauthapi/config/AuthorizationSessionKeyConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [de.adorsys.opba.tppauthapi.config.AuthorizationSessionKeyConfig$AuthorizationSessionKeyFromHttpRequest]: Factory method 'getAuthorizationSessionKeyFromHttpRequest' threw exception; nested exception is java.lang.IllegalStateException: Authorization-Session-Key cookie is missing
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:656)
....

This does not tell me that a client made a call without providing a cookie. Further I see full stack of exception. In regular use case. Yes it would work, but log will be polluted by exception. I would prefer my soltuion.

Reply:
Just implement controller advice and throw specialized exception.

Reply:
yep, but it still is an exception. with stack. though it comes with regular work flow. let us talk on monday. I do not want to have exception, just because client is stupidly asking for cookierefreshment. I just want warning message without stack.

Reply:
A proper solution for this problem is WebFilter that filters out requests without cookies. If request processing is bound to fail it must fail as soon as possible and not leak deeper to business layer